### PR TITLE
Custom errors

### PR DESCRIPTION
**Pull request by ostap0207**
Info in the commits.

CHAN-1440

### CODE REVIEW

**Comment by ostap0207** [comment]
Here is an example of how it will look like https://github.com/salemove/phone-number-manager/pull/229

**Review by urmastalimaa — APPROVED** [review]
Seems to be a bit janky, from the example it's hard to understand what the benefit of using this whole glia errors actually is, when there needs to be such a dance. Probably this is just ironing out an edge-case, and I'm missing the light at the end of the tunnel.

**Comment by ostap0207** [comment]
First few lines in the readme kinda explain that:
> The library provides a list of errors that are used in Glia REST API and utilities to easily construct them.

> It is used by Glia developers to map application specific errors to one of standardized Glia errors.

File: lib/glia/errors/mapper.rb [hunk 1]
```diff
@@ -120,26 +120,31 @@ module Mapper
         'must be a valid email' => INVALID_FORMAT_ERROR
       }.freeze
 
-      def self.from_dry_validation_result(output, messages)
+      def self.from_dry_validation_result(output, messages, custom_error_map = {})
+        error_map = ERROR_MAP.merge(custom_error_map)
+        from_dry_validation_result_rec(output, messages, error_map)
```
Comment (SteamRiC) [inline lib/glia/errors/mapper.rb:125]: what does `rec` stand for?

Comment (ostap0207) [reply lib/glia/errors/mapper.rb:125]: recursive

**Review by SteamRiC — APPROVED** [review]

**Comment by ostap0207** [comment]
!merge

**Pull request merged**